README.md

```diff
@@ -1,2 +1,114 @@
 # java-globaltracer
 Global ThreadLocal Span for implicit propagation, delegating to another Tracer implementation.
+
+This library provides two main utiltiy classes; `GlobalTracer` and `GlobalSpanManager`.
+The first is intended for used by application code and library developers,
+where the latter is more intended for boundary filters and `Tracer` implementations.
+
+## GlobalTracer
+This utility class has three main purposes:
+ 1. Standardized static access to the configured global tracer in the application:
+    `GlobalTracer.tracer()`.
+    This method never returns a `null` reference, but at the very least a `NoopTracer` instance.
+ 2. Providing a Java SPI ServiceLoader support for compatible `Tracer` implementations
+    that contain a `META-INF/services/io.opentracing.Tracer` entry.
+    In case there are no (or multiple) Tracer implementations, 
+    a configured tracer can be registered explicitly by calling `GlobalTracer.register()`.
+ 3. Wrapping the configured global tracer implementation so all `Span` instances
+    created from it are automatically registered as _active span_ and deregistered
+    when they finish.
+
+## GlobalSpanManager
+This utility class provides the following functionality.
+ 1. The `GlobalSpanManager.activate()` static method that is automatically called from
+    the `GlobalTracer` instance. 
+ 2. The `Closable` result from this `activate()` method will deactviate the given `Span`
+    if needed. However, the `Spans` returned by the `GlobalTracer` will also make sure
+    they get deactivated whenever they're finished or closed.
+ 3. A method `GlobalSpanManager.activeSpan()` that will return the _active span_ 
+    from anywhere in the code. This may be useful for libraries that wish to add tags
+    or log information in the active span.
+ 4. It contains a default implementation that uses `ThreadLocal` storage to manage
+    the globally _active span_, activating and deactivating it at the appropriate
+    moments (starting / closing spans etc).
+ 5. `Tracer` implementations or applications that require more customized span management
+    can provide their own subclass of `GlobalSpanManager` and either register it 
+    programmatically through the `GlobalSpanManager.registerInstance()` method 
+    or automatically by the Java SPI ServiceLoader via a 
+    `META-INF/services/io.opentracing.global.GlobalSpanManager` service definition.
+
+## How to use this library
+This section tries to give some basic examples on how to use the Global tracer and
+identify some caveats that need to be taken into account.
+
+### From miscelaneous application / library code
+To add tracing functionality to code that cannot manually propagate `Tracer` or `Span` instances
```

Comment:
putting a concrete example could help. Best I can think of is the mysql interceptor (or anything else which has a static context concern)